test/iterator-test.js

```diff
@@ -152,6 +152,24 @@ make('iterator seek before next has completed', function (db, t, done) {
   t.ok(error, 'had error from seek() before next() has completed')
 })
 
+make('close db with open iterator', function (db, t, done) {
+  var ite = db.iterator()
+  var cnt = 0
+  ite.next(function loop(err, key, value) {
+    if(cnt++ === 0)
+      t.error(err, 'no error from next()')
+    else
+      t.ok(err, 'had error from next() after end()')
```

Comment:
Maybe check the error message here instead, e.g. `t.equal(err.message, 'iterator has ended')`